Mujoco get_states and set_states inconsistency

## How to reproduce
```bash
python metasim/test/state_consistency.py --sim=mujoco
```

which raise the following error:
```
Traceback (most recent call last):
  File "/home/fs/cod/RoboVerse/metasim/test/state_consistency.py", line 151, in <module>
    assert_close(states[i]["objects"]["cube"]["pos"], init_states[i]["objects"]["cube"]["pos"])
  File "/home/fs/cod/RoboVerse/metasim/test/state_consistency.py", line 143, in assert_close
    assert torch.allclose(a, b, atol=atol), f"a: {a} != b: {b}"
AssertionError: a: tensor([0., 0., 0.]) != b: tensor([ 0.3000, -0.2000,  0.0500])
```

@charlietcheng  Could you take a look at this issue?

After this bug fixed, the following lines could be deleted:
https://github.com/RoboVerseOrg/RoboVerse/blob/1518c6f8ff8a4e869b39c0865a0d1efb10cdfbc1/metasim/sim/env_wrapper.py#L93-L95

Please make sure to also delete it and pass your test in your PR.

--- a/metasim/sim/env_wrapper.py
+++ b/metasim/sim/env_wrapper.py
@@ -90,7 +90,7 @@ def GymEnvWrapper(cls: type[THandler]) -> type[EnvWrapper[THandler]]:
             self._episode_length_buf[env_ids] = 0
             if states is not None:
                 self.handler.set_states(states, env_ids=env_ids)
-            if self.handler.scenario.sim in ["mujoco", "isaacgym"]:
+            if self.handler.scenario.sim in ["isaacgym"]:
                 ## HACK
                 self.handler.simulate()
             self.handler.checker.reset(self.handler, env_ids=env_ids)

--- a/metasim/sim/mujoco/mujoco.py
+++ b/metasim/sim/mujoco/mujoco.py
@@ -351,6 +351,7 @@ class MujocoHandler(BaseSimHandler):
         for obj_name, obj_state in states_flat[0].items():
             self._set_root_state(obj_name, obj_state, zero_vel)
             self._set_joint_state(obj_name, obj_state, zero_vel)
+        self.physics.forward()
 
     def _disable_robotgravity(self):
         gravity_vec = np.array([0.0, 0.0, -9.81])

